kn/knative-dockerfile-deploy/README.md

```diff
@@ -0,0 +1,97 @@
+# Git Source with Dockerfile to Knative Service
+
+This documents the example Pipeline to build the source code with a Dockerfile in
+the git repo and deploy it as Knative Service.
+It uses [buildah task](../../buildah/README.md) for building the source code and
+[kn task](../README.md) to create or update a Knative Service.
+
+## Prerequisites:
+
+1. Latest Tekton Pipelines [install](https://github.com/tektoncd/pipeline/blob/master/docs/install.md)ed.
+2. `kubectl` CLI [install](https://kubernetes.io/docs/tasks/tools/install-kubectl/)ed.
+3. `tkn` CLI [install](https://github.com/tektoncd/cli#installing-tkn)ed.
+4. User account exists at a container registry (e.g. [quay.io](https://quay.io))
+5. A ServiceAccount to enable access to perform the required operations in the Pipeline,
+   we'll configure one in following section.
+
+### One time setup:
+
+1 - Create a sample namespace `tkn-kn`, we'll reference this namespace in the subsequent operations.
+```bash
+kubectl create namespace tkn-kn
+```
+
+2 - Create a `docker-registry` type secrets for pushing/pulling the built container images.
+```bash
+kubectl create secret docker-registry container-registry --docker-server=<DOCKER-REGISTRY> --docker-username=<USERNAME> --docker-password=<PASSWORD> --docker-email=<EMAIL>
+```
+#### Note:
+- If you are using `docker.io`: Push to (a new private repo at) `docker.io` via buildah and pull via `imagePullSecrets` does not work!
+  Please [create a new](https://hub.docker.com/repository/create) empty public repository for this tutorial and refer it in subsequent steps.
+- If you are using `quay.io`: Push to `quay.io` via buildah and pull via `imagePullSecrets` works well. [Get](https://quay.io/signin/) an account created at quay.
```

Comment:
If you wanted to tell the user to have a default `imagePullSecrets`maybe you want to link this https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/ in there?

Reply:
Sure, added a line in above section where secrets are created as:

 More about secrets for [`interacting with a private registry`](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).